server/backend/rpcserver.go

```diff
@@ -69,6 +76,46 @@ func (s *CampusServer) GetNewsSources(ctx context.Context, _ *emptypb.Empty) (ne
 	return &pb.NewsSourceArray{Sources: resp}, nil
 }
 
+// SearchRooms returns all rooms that match the given search query.
+func (s *CampusServer) SearchRooms(ctx context.Context, req *pb.SearchRoomsRequest) (*pb.SearchRoomsReply, error) {
+	if err := s.checkDevice(ctx); err != nil {
+		return nil, err
+	}
+	var res []struct { // struct to scan database query into
```

Comment:
Please add a check if the query has anything in it and return early to save a few cycles.